proofs/tree.go

```diff
@@ -280,6 +280,41 @@ func NewDocumentTree(proofOpts TreeOptions) DocumentTree {
 	}
 }
 
+// NewDocumentTree returns a DocumentTree with that has a root hash set.
+// It can be used to validate proofs but not for creating any.
+func NewDocumentTreeWithRootHash(proofOpts TreeOptions, rootHash []byte) DocumentTree {
```

Comment:
Can we reuse the `NewDocumentTree` function so we do not duplicate much code? Like:
```
func NewDocumentTreeWithRootHash(proofOpts TreeOptions, rootHash []byte) DocumentTree {
  documentTree := NewDocumentTree(proofOpts)
  documentTree.rootHash = rootHash
  return documentTree
}
```

Reply:
Yeah. Makes sense. I'd do that. Thanks for pointing it out 🙂